KOGITO-7954 : SWF Editor - Some basic automated integration tests

inodeman opened the pull request:
Hi @romartin can you review this PR, it has the integration tests requested. Thanks

domhanak commented on packages/serverless-workflow-diagram-editor/sw-editor/sw-editor-kogito-app/src/test/java/org/kie/workbench/common/stunner/sw/client/selenium/SWAPITests.java:
```diff
@@ -0,0 +1,397 @@
+     * @return Text content of the file
+     * @throws IOException
+     */
+    private String loadResource(final String filename) throws IOException {
```
Duplicated method from SWEditorSeleniumIT.

I suggest creating SWEditorBaseSeleniumIT where this method will be and this class should inherit from the base.

inodeman replied:
Hi @domhanak will move to base class

domhanak commented on packages/serverless-workflow-diagram-editor/sw-editor/sw-editor-kogito-app/src/test/java/org/kie/workbench/common/stunner/sw/client/selenium/SWAPITests.java:
```diff
@@ -0,0 +1,397 @@
+                .collect(Collectors.joining(""));
+    }
+
+    private WebDriverWait waitOperation() {
```
Duplicated method from SWEditorSeleniumIT.

I suggest creating SWEditorBaseSeleniumIT where this method will be and this class should inherit from the base.

inodeman replied:
Hi @domhanak will move to base class

domhanak commented on packages/serverless-workflow-diagram-editor/sw-editor/sw-editor-kogito-app/src/test/java/org/kie/workbench/common/stunner/sw/client/selenium/SWAPITests.java:
```diff
@@ -0,0 +1,397 @@
+    }
+
+    @Before
+    public void openSWEditor() {
```
Duplicated method from SWEditorSeleniumIT.

I suggest creating SWEditorBaseSeleniumIT where this method will be and this class should inherit from the base.

inodeman replied:
Hi @domhanak will move to base class

domhanak commented on packages/serverless-workflow-diagram-editor/sw-editor/sw-editor-kogito-app/src/test/java/org/kie/workbench/common/stunner/sw/client/selenium/SWAPITests.java:
```diff
@@ -0,0 +1,397 @@
+        setContent(expected);
+        waitCanvasPanel();
+
+        final Object result = ((JavascriptExecutor) driver).executeScript(String.format(GET_NODE_IDS));
```
This should be extracted to a method so it can be reused in other tests.
Example:
```
// Setup
JsCanvasHelper jsHelper = new JsCanvasHelper((JavascriptExecutor) driver);
// Calls
final ArrayList<String> nodeIDList = jsHelper.getNodeIds();
```

inodeman replied:
Hi @domhanak will extract to helper class

domhanak commented on packages/serverless-workflow-diagram-editor/sw-editor/sw-editor-kogito-app/src/test/java/org/kie/workbench/common/stunner/sw/client/selenium/SWAPITests.java:
```diff
@@ -0,0 +1,397 @@
+        assertThat(nodeIds.size()).isEqualTo(2);
+
+        for (String string : nodeIds) {
+            Object backgroundColorResult = ((JavascriptExecutor) driver).executeScript(String.format(GET_BACKGROUND_COLOR.replace("%s", string)));
+            Object borderColorResult = ((JavascriptExecutor) driver).executeScript(String.format(GET_BORDER_COLOR.replace("%s", string)));
```
Should be extracted to some `JsCanvasHelper` class.

inodeman replied:
Hi @domhanak will extract to helper class

domhanak commented on packages/serverless-workflow-diagram-editor/sw-editor/sw-editor-kogito-app/src/test/java/org/kie/workbench/common/stunner/sw/client/selenium/SWAPITests.java:
```diff
@@ -0,0 +1,397 @@
+        List<String> nodeIds = (ArrayList<String>) result;
+        assertThat(nodeIds.size()).isEqualTo(2);
+
+        Object location1 = ((JavascriptExecutor) driver).executeScript(String.format(GET_LOCATION.replace("%s", nodeIds.get(0))));
+        Object absolutelocation1 = ((JavascriptExecutor) driver).executeScript(String.format(GET_ABSOLUTE_LOCATION.replace("%s", nodeIds.get(0))));
```
Should be extracted to `JsCanvasHelper`

inodeman replied:
Hi @domhanak will extract to helper class

domhanak commented on packages/serverless-workflow-diagram-editor/sw-editor/sw-editor-kogito-app/src/test/java/org/kie/workbench/common/stunner/sw/client/selenium/SWAPITests.java:
```diff
@@ -0,0 +1,397 @@
+        List<String> nodeIds = (ArrayList<String>) result;
+        assertThat(nodeIds.size()).isEqualTo(2);
+
+        Object dimension1 = ((JavascriptExecutor) driver).executeScript(String.format(GET_DIMENSIONS.replace("%s", nodeIds.get(0))));
```
Should be xtracted to `JsCanvasHelper`

inodeman replied:
Hi @domhanak will extract to helper class

inodeman commented on packages/serverless-workflow-diagram-editor/sw-editor/sw-editor-kogito-app/src/test/java/org/kie/workbench/common/stunner/sw/client/selenium/SWAPITests.java:
```diff
@@ -0,0 +1,397 @@
+                .isNotNull();
+    }
+
+    private void setContent(final String xml) {
```
Hi @domhanak will move to base class

inodeman commented:
Hi @domhanak @romartin I extracted common methods from Selenium test classes to a base class and refactored the code for the JS Api calls as requested by Dominik, code is now easier to follow and shorter, thanks, let me know if ok

domhanak commented on packages/serverless-workflow-diagram-editor/sw-editor/sw-editor-kogito-app/src/test/java/org/kie/workbench/common/stunner/sw/client/selenium/SWAPITests.java:
```diff
@@ -0,0 +1,290 @@
+        assertThat(isConnection).isEqualTo(true);
+    }
+
+    public class JsCanvasHelper {
```
I guess this is the only thing now. What about we make this a separate class and init it by default in base class so it can be use any follow-up tests that extend the base.

I think it makes sense, what do you think?

inodeman replied:
Hi @domhanak sure, will do that

domhanak commented on packages/serverless-workflow-diagram-editor/sw-editor/sw-editor-kogito-app/src/test/java/org/kie/workbench/common/stunner/sw/client/selenium/SWAPITests.java:
```diff
@@ -0,0 +1,290 @@
+        for (String uuid : nodeIds) {
+            String backgroundColor = jsHelper.getGetBackgroundColor(uuid);
+            String borderColor = jsHelper.getBorderColor(uuid);
+            ;
```
extra semicolon?

inodeman replied:
Hi @domhanak removed

domhanak commented on packages/serverless-workflow-diagram-editor/sw-editor/sw-editor-kogito-app/src/test/java/org/kie/workbench/common/stunner/sw/client/selenium/SWAPITests.java:
```diff
@@ -0,0 +1,290 @@
+
+        /// None
+        jsHelper.applyState(uuid, "none");
+        Thread.sleep(2000); // Otherwise sometime it applies the state and sometimes it does not finish
```
Maybe the `Thread.sleep()` should be executed in `applyState()` ?

inodeman replied:
Hi @domhanak will move to helper method

domhanak commented:
@inodeman Thank you I noticed some areas that can be improved so I reported a ticket: https://issues.redhat.com/browse/KOGITO-8205 I will work on it later.
I am approving.

domhanak reviewed (approved):
@romartin missing review from your side

romartin reviewed (approved):
It looks good to me! Thanks @inodeman
